Backend/services/notificationService.js

```diff
@@ -83,22 +88,46 @@ const sendNotification = async (notificationData, io) => {
           createdAt: new Date(), // Update the timestamp
           isImportant,
           content: `@${notificationData.usernameTarget} followed you.`, // Update the content
+          isPending: true,
+          isRead: false,
         },
       });
     }
   } else {
     const notification = {
       ...notificationData,
       isImportant,
+      isPending: true,
     };
 
     savedNotification = await prisma.notification.create({
       data: notification,
     });
   }
 
+  const userActivePeriods = await prisma.userActivePeriods.findUnique({
+    where: { userId },
+  });
+
+  if (!userActivePeriods) {
+    if (userSockets.length > 0) {
+      userSockets[0].emit("notification", savedNotification);
+    }
+    await prisma.notification.update({
+      where: { id: savedNotification.id },
+      data: { isPending: false },
+    });
+  }
+  await scheduleNotifications(userId, savedNotification, io);
+  // Emit pending notification count
+  const pendingCount = await prisma.notification.count({
+    where: {
+      userId,
+      isPending: true,
+    },
+  });
   if (userSockets.length > 0) {
-    userSockets[0].emit("notification", savedNotification);
+    userSockets[0].emit("pendingNotificationCount", pendingCount);
```

Comment:
should this all be in an else statement? do we still want to do these if the user doesn't have an active period?

Reply:
thats true